Modules/posixmodule.c

```diff
@@ -12916,14 +12916,15 @@ _pystatvfs_fromstructstatfs(PyObject *module, struct statfs st) {
 
     _Static_assert(sizeof(st.f_blocks) == sizeof(long long), "assuming large file");
 
-#define SET_ITEM(v, index, item)                   \
-    do {                                           \
-        if (item == NULL) {                        \
-            Py_DECREF(v);                          \
-            return NULL;                           \
-        }                                          \
-        PyStructSequence_SET_ITEM(v, index, item); \
-    } while (0)                                    \
+#define SET_ITEM(V, INDEX, EXPR)                    \
+    do {                                            \
+        PyObject *obj = (EXPR);                     \
+        if (obj == NULL || PyErr_Occurred()) {      \
```

Comment:
Checking for `PyErr_Occurred()` is not needed.

```suggestion
        if (obj == NULL) {      \
```